aiida/orm/nodes/data/code/containerized.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""
+Data plugins representing an executable code to be run in a container.
+These plugins are directly analogous to the ``InstalledCode`` and ``PortableCode`` plugins, except that the executable
+is present inside of a container. For the ``InstalledContainerizedCode`` the executable is expected to already be
+present inside a container that is available on the target computer. With the ``PortableContainerizedCode`` plugin, the
+target executable will be stored in AiiDA's storage, just as with the ``PortableCode`` and when launched, the code will
+be copied inside the container on the target computer and run inside the container.
+"""
+from __future__ import annotations
+
+import click
+
+from aiida.common.lang import type_check
+
+from .abstract import AbstractCode
+from .installed import InstalledCode
+from .portable import PortableCode
+
+__all__ = (
+    'InstalledContainerizedCode',
+    'PortableContainerizedCode',
+)
+
+
+class Containerized(AbstractCode):
+    """Containerized Mixin for code that can run on container"""
+    _KEY_ATTRIBUTE_ENGINE_COMMAND: str = 'engine_command'
+    _KEY_ATTRIBUTE_IMAGE: str = 'image'
```

Comment:
I think I already suggested this in a review of one of the earlier PRs. I think making it explicit that it is the _name_ of the image is clearer.
```suggestion
    _KEY_ATTRIBUTE_IMAGE_NAME: str = 'image_name'
```